### DISCUSSION TITLE
Too slow to persist and fetch jobs from a central postgresql store - alternatives?

### DISCUSSION THREAD

**Discussion by NixBiks** [agronholm/apscheduler#812] [General]
I have a central datastore, postgresql, where I want to keep all the info about my schedules. I want multiple workers in different regions to all run all jobs. So when I start a worker starts I want the worker to pick up all schedules, calculate when to execute and execute - the workers can keep all the info in memory except it needs to pick up all schedules from the central datastore AND it needs to listen for any changes to schedules.

Is something like this doable in v4?

**Comment by agronholm** [comment]
This isn't exactly how it works on v4, as having workers keep all the schedules in memory isn't really scalable. But other than that this sounds like what v4 does.

**Reply by agronholm** [reply]
What kind of requirements do you have that a PostgreSQL data store would be too slow for?

**Reply by NixBiks** [reply]
I've come to the conclusion that it's out of scope for APScheduler. I went with a different approach where I've "templated" the schedules for various workers. When a worker then boots up it'll add the schedules to a local storage apscheduler engine. And then I have a websocket listening for db changes which fires a callback to APScheduler. Works pretty good
